frontend/src/shared/helpers.tsx

```diff
@@ -512,6 +515,83 @@ function setupEditableSampleFields(samplesColDefs: ColDef[]) {
   });
 }
 
+export function prepareCohortDataForAgGrid(
+  cohortsListQueryResult: CohortsListQuery,
+  filterModel: IServerSideGetRowsRequest["filterModel"]
+) {
+  const { cohorts, cohortsConnection } = cohortsListQueryResult;
+
+  let newCohorts = cohorts.map((cohort) => {
+    const {
+      cohortId,
+      hasCohortSampleSamplesConnection: samplesConnection,
+      hasCohortSampleSamples: samples,
+      hasCohortCompleteCohortCompletes: cohortCompletes,
+    } = cohort;
+
+    const totalSamples = samplesConnection?.totalCount;
+
+    const allSamplesBilled =
+      samples.length > 0 &&
+      samples?.every((sample) => {
+        return sample.hasTempoTempos?.[0].billed === true;
+      });
+    const billed = allSamplesBilled === true ? "Yes" : "No";
+
+    const initialCohortDeliveryDate = cohortCompletes?.slice(-1)[0]?.date;
+
+    const latestCohortDeliveryDate = cohortCompletes?.[0];
+    const {
+      date: completeDate,
+      endUsers,
+      pmUsers,
+      projectTitle,
+      projectSubtitle,
+      status,
+      type,
+    } = latestCohortDeliveryDate ?? {};
+
+    return {
+      cohortId,
+      totalSamples,
+      billed,
+      initialCohortDeliveryDate: formatCohortRelatedDate(
+        initialCohortDeliveryDate
+      ),
+      completeDate: formatCohortRelatedDate(completeDate),
+      endUsers,
+      pmUsers,
+      projectTitle,
+      projectSubtitle,
+      status,
+      type,
+    };
+  });
+
+  let newCohortsConnection = { ...cohortsConnection };
+
+  const selectedAll = JSON.stringify(filterModel) === "{}";
+  if (!selectedAll) {
+    const selectedValues = filterModel.billed.values;
+    const selectedNone = selectedValues.length === 0;
+
+    if (selectedNone) {
+      newCohorts = [];
+      newCohortsConnection.totalCount = 0;
+    } else {
+      newCohorts = newCohorts.filter(
+        (cohort) => cohort.billed === selectedValues[0]
+      );
+      newCohortsConnection.totalCount = newCohorts.length;
+    }
+  }
```

Comment:
Filter data based on what users want to filter by. Also calculate the new number of cohorts based on newly filtered data.

Reference the examples [here](https://github.com/mskcc/smile-dashboard/pull/120/files#r1581458274) on what `filterModel` looks like.